File: 0015 三数之和/main2.py
from typing import List
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        tmpSet = []
        nums.sort()
        lasti = None
        for i in range(len(nums)-2):

            valuei = nums[i]
            if(lasti==nums[i]):
                continue

            li = i+1
            ri = len(nums)-1                

            while(li != ri):
                # print(nums, valuei,nums[li], nums[ri])
                if(0 == nums[li]+nums[ri]+valuei):
                    tmpSet.append([valuei, nums[li], nums[ri]])
                    ri -= 1
                elif(0 <= nums[li]+nums[ri]+valuei):
                    ri -= 1
                else:
                    li += 1
            lasti=nums[i]
        return tmpSet

File: 0080 删除有序数组中的重复项 II/main2.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int count=1;
        int idx=1;
        for(int i=1;i<nums.size();i++){
            if( nums[i] == nums[i-1]){
                count++;
            }
            else{
                count = 1;
            }
            if(count <= 2)
                nums[idx++]=nums[i];
        }
        return idx;
    }
};